app/scenes/Document/components/DataLoader.js

```diff
@@ -70,6 +72,27 @@ class DataLoader extends React.Component<Props> {
   }
 
   onSearchLink = async (term: string) => {
+    if (isInternalUrl(term)) {
+      // search for exact internal document
+      const slug = parseDocumentSlugFromUrl(term);
+      try {
+        const document = await this.props.documents.fetch(slug);
+        return [
+          {
+            title: document.title,
+            url: document.url,
+          },
+        ];
+      } catch (err) {}
+    } else {
+      try {
+        // don't search at all if term is a non-internal url
+        new URL(term);
+        return;
+      } catch (err) {}
+    }
```

Comment:
The try catch look is a little odd, would love some syntax advice

Reply:
I'd consider just dropping this case to be honest – I'm worried that genuine search terms could be successfully parsed as valid URL's and it doesn't do too much harm to just send to the document search – it might even turn up a relevant doc.